config_src/drivers/nuopc_cap/mom_surface_forcing_nuopc.F90

```diff
@@ -544,6 +543,25 @@ subroutine convert_IOB_to_fluxes(IOB, fluxes, index_bounds, Time, valid_time, G,
     fluxes%sw(i,j) = fluxes%sw_vis_dir(i,j) + fluxes%sw_vis_dif(i,j) + &
                      fluxes%sw_nir_dir(i,j) + fluxes%sw_nir_dif(i,j)
 
+    ! enthalpy terms
+    if (associated(IOB%hrofl)) &
+      fluxes%heat_content_lrunoff(i,j) = US%W_m2_to_QRZ_T * IOB%hrofl(i-i0,j-j0) * G%mask2dT(i,j)
+
+    if (associated(IOB%hrofi)) &
+      fluxes%heat_content_frunoff(i,j) = US%W_m2_to_QRZ_T * IOB%hrofi(i-i0,j-j0) * G%mask2dT(i,j)
+
+    if (associated(IOB%hrain)) &
+      fluxes%heat_content_lprec(i,j)   = US%W_m2_to_QRZ_T * IOB%hrain(i-i0,j-j0) * G%mask2dT(i,j)
+
+    if (associated(IOB%hsnow)) &
+      fluxes%heat_content_fprec(i,j)   = US%W_m2_to_QRZ_T * IOB%hsnow(i-i0,j-j0) * G%mask2dT(i,j)
+
+     if (associated(IOB%hevap)) &
+      fluxes%heat_content_evap(i,j)    = US%W_m2_to_QRZ_T * IOB%hevap(i-i0,j-j0) * G%mask2dT(i,j)
```

Comment:
If I am following the code correctly, IOB%hevap is allocated at all times when cesm_coupled is True. However, fluxes%heat_content_evap gets allocated only if ENTHALPY_FROM_COUPLER is set to True, which is problematic. I think IOB%hevap should also be allocated only when ENTHALPY_FROM_COUPLER is set to True. This may apply to other heat_content components too. @gustavo-marques ?

Reply:
Thanks for pointing this out. I could not find an easy way to control the allocation of the enthalpy terms in the IOB type. Instead, I added a conditional statement for setting fluxes%heat_content_* in this subroutine. I tested this modification by running a BMOM case with ENTHALPY_FROM_COUPLER = False and there were no issues.